.github/workflows/sdk.yaml

```diff
@@ -32,20 +32,17 @@ jobs:
         working-directory: ./sdk
     steps:
       - name: Check out code
-        uses: actions/checkout@v3
+        uses: actions/checkout@v4
 
       - name: Setup Python ${{ matrix.python-version }}
-        uses: actions/setup-python@v2
+        uses: actions/setup-python@v5
         with:
           python-version: ${{ matrix.python-version }}
-
-      - name: Cache pip dependencies
-        uses: actions/cache@v2
-        with:
-          path: ~/.cache/pip
-          key: |
-            pip-${{ hashFiles('sdk/requirements.txt') }}-${{ hashFiles('sdk/requirements.dev.txt') }}-${{ matrix.python-version }}
-          restore-keys: pip-
+          cache: pip
+          cache-dependency-path: |
+            sdk/requirements.txt
+            sdk/requirements.dev.txt
+            engines/pyfunc-ensembler-service/requirements.dev.txt
```

Comment:
Example of the cache being restored successfully: https://github.com/caraml-dev/turing/actions/runs/9804032057/job/27071120655#step:3:22